File: problem20.php
<?php
//solution: 648
echo eulerProblem20() . "\n";

function eulerProblem20() {
    $str = factorial(100);
    
    $sum = 0;
    for($i = 0; $i < strlen($str); ++$i) {
        $sum += $str{$i};
    }
    
    return $sum;
}

function factorial($num) {
    if($num == 1) {
        return 1;
    }
    
    return bcmul($num, factorial($num-1));
}

?>

File: python/problem20.py
#!/usr/bin/python
#solution: 648

def factorial(n):
    if n == 2:
        return 2
    return n * factorial(n - 1)

def euler_problem_20():
    return reduce(lambda x, y: int(x) + int(y), str(factorial(100)))

print euler_problem_20()
